src/reducer.js

```diff
@@ -434,10 +444,44 @@ const onUpdateHeight = (state, action) => {
   return state;
 };
 
+const onSetWallet = (state, action) => {
+  if (state.wallet && state.wallet.state !== 0) {
+    // Wallet was not closed
+    state.wallet.stop();
```

Comment:
Does it return a promise? Should we wait for it?

Reply:
It does not return a promise.